### DISCUSSION TITLE
Support to specifing commit email for `Forge-Merge`

### DISCUSSION THREAD

**Discussion by Thaodan** [magit/forge#620] [Ideas]
I noticed that when merging using the API with `forge-merge` that the default email of the account that the user has on the target forge is used. 
Forges such as e.g. Github allow to select which email is used to merge the pull request using the website/API.

**Comment by tarsius** [comment]
I don't see that here https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#merge-a-pull-request.

**Comment by Thaodan** [comment]
Could it be part of a private API? The Github Ui does support it
somehow.

**Reply by tarsius** [reply]
I cannot find this feature in the web interface either. The way you describe it, I would expect there to be a drop-down menu `Merge using identity` next to `Merge pull request`.

If on the other hand, you are talking about a feature that involves going to the repository's settings and setting the committer identity for all future merges there, then I would assume that setting would also take effect when merging using the API.

**Reply by Thaodan** [reply]
The feature is explained here:
https://github.blog/changelog/2022-09-15-git-commit-author-shown-when-squash-merging-a-pull-request/

I don't have a repository where I could take a screenshot right now.

**Reply by tarsius** [reply]
I guess one is only given a choice if the commits to be squashed were authored by multiple people, and I don't have such a pull-request at hand to confirm that this option becomes available then.

Anyway, the API does not offer this, so there's nothing I can do about it.
